core/common/src/rpc/web/BentleyCloudRpcManager.ts

```diff
@@ -18,20 +18,46 @@ import { RpcRoutingToken } from "../core/RpcRoutingToken";
 /* eslint-disable deprecation/deprecation */
 
 /** Initialization parameters for BentleyCloudRpcConfiguration.
- * @internal
+ * @public
  */
 export interface BentleyCloudRpcParams {
   /** Identifies the remote server that implements a set of RpcInterfaces. Note that the ID of the remote server is not a URI or hostname. It is a string that matches a key in the orchestrator's app registry. */
   info: OpenAPIInfo;
-  /** The protocol for Bentley cloud RPC interface deployments */
-  protocol?: typeof BentleyCloudRpcProtocol;
   /** The URI of the orchestrator that will route requests to the remote RpcInterface server. If not supplied, this default to the origin of the Web page. This is required only when calling initializeClient and only if the server is not the origin of the Web page. */
   uriPrefix?: string;
+}
+
+/** Internal initialization parameters for BentleyCloudRpcConfiguration.
+ * @internal
+ */
+export interface BentleyCloudRpcInternalParams extends BentleyCloudRpcParams {
+  /** The protocol for Bentley cloud RPC interface deployments */
+  protocol?: typeof BentleyCloudRpcProtocol;
   /** Handler for RPC request events. */
   pendingRequestListener?: RpcRequestEventHandler;
 }
 
-/** Operating parameters for Bentley cloud RPC interface deployments.
+/** Holds initialization parameters for BentleyCloudRpcConfiguration.
+ * @public
+ */
+export class BentleyCloudRpcParamsHolder {
```

Comment:
what's the purpose of this class?

Reply:
To wrap the internal params that a consumer might still be using.

I moved BentleyCloudRpcParam’s two optional params to an internal interface, since publicizing them would have rippled through almost the entire RPC api.

Reply:
A public class without a public constructor doesn't make sense. You're doing crazy workarounds to avoid (strictly speaking) violating our policies but this only makes the user experience much worse.

Reply:
The alternative is to publicize basically the entire rpc system. So, that’s the value of this workaround.

I think it’s the right trade off.

Reply:
so you think having a weird workaround that no one will understand is better than just leaving this public and _not_ rippling it through all RPC code?

Reply:
can we get a way with making this `@beta`?
not a fan of introducing temporary solutions and marking them as `@public`

Reply:
That sounds like a good idea. I will try it…

Reply:
I switched them to beta.

Reply:
If we're going to make it `beta` any reason we can't stick with the current API without the wrapper and then mark the field we don't want to expose as `internal`? Most people won't use it but for those that are doing it today (I know I've used the protocol in a few different cases).

Also, I've seen in a few other files that were changes we need to use the protocol to get the port for setting up the backend. Do we have an alternative for that?